<?xml version="1.0"?>

<robot name="puzzlebot">
    <link name="base_footprint">
        <inertial>
            <mass value="0.01"/>
            <origin xyz="0 0 0" rpy="0 0 0"/>
            <inertia
            ixx="0.0001" ixy="0.0" ixz="0.0"
            iyy="0.0001" iyz="0.0"
            izz="0.0001"/>
        </inertial>
    </link>

    <link name="base_link">
        <visual>
            <origin xyz="0. 0. -0.0025" rpy="0. 0. 0."/>
            <geometry>
                <mesh filename="package://mlr_nav2_puzzlebot/meshes/Puzzlebot_Jetson_Lidar_Edition_Base.stl"/>
            </geometry>
            <material name="base_color">
                <color rgba="1.0 1.0 1.0 1.0"/>
            </material>
        </visual>

        <collision>
            <geometry>
                <mesh filename="package://mlr_nav2_puzzlebot/meshes/Puzzlebot_Jetson_Lidar_Edition_Base.stl"/>
            </geometry>
        </collision>

        <inertial>
            <mass value="2.0"/>
            <origin xyz="0 0 0" rpy="0 0 0"/>
            <inertia
            ixx="0.0883" ixy="0.0" ixz="0.0"
            iyy="0.032" iyz="0.0"
            izz="0.1070"/>
        </inertial>
    </link>

    <link name="wheel_l">
        <visual>
            <origin xyz="0. 0. 0." rpy="1.57 0. 0."/>
            <geometry>
                <mesh filename="package://mlr_nav2_puzzlebot/meshes/Puzzlebot_Wheel.stl"/>
            </geometry>
            <material name="wheel_color">
                <color rgba="1.0 1.0 1.0 1.0"/>
            </material>
        </visual>

        <collision>
            <geometry>
                <mesh filename="package://mlr_nav2_puzzlebot/meshes/Puzzlebot_Wheel.stl"/>
            </geometry>
        </collision>

        <inertial>
            <mass value="0.2"/>
            <origin xyz="0 0 0" rpy="1.57 0 0"/>
            <inertia
            ixx="0.0007" ixy="0.0" ixz="0.0"
            iyy="0.0007" iyz="0.0"
            izz="0.00004"/>
        </inertial>
    </link>

    <link name="wheel_r">
        <visual>
            <origin xyz="0. 0. 0." rpy="1.57 0. 3.14"/>
            <geometry>
                <mesh filename="package://mlr_nav2_puzzlebot/meshes/Puzzlebot_Wheel.stl"/>
            </geometry>
            <material name="wheel_color">
                <color rgba="1.0 1.0 1.0 1.0"/>
            </material>
        </visual>

        <collision>
            <geometry>
                <mesh filename="package://mlr_nav2_puzzlebot/meshes/Puzzlebot_Wheel.stl"/>
            </geometry>
        </collision>

        <inertial>
            <mass value="0.2"/>
            <origin xyz="0 0 0" rpy="1.57 0 0"/>
            <inertia
            ixx="0.0007" ixy="0.0" ixz="0.0"
            iyy="0.0007" iyz="0.0"
            izz="0.00004"/>
        </inertial>
    </link>

    <link name="caster">
        <visual>
            <origin xyz="0. 0. -0.0155" rpy="0. 0. 0."/>
            <geometry>
                <mesh filename="package://mlr_nav2_puzzlebot/meshes/Puzzlebot_Caster_Wheel.stl"/>
            </geometry>
            <material name="caster_color">
                <color rgba="0.6 0.6 0.6 1.0"/>
            </material>
        </visual>

        <collision>
            <geometry>
                <mesh filename="package://mlr_nav2_puzzlebot/meshes/Puzzlebot_Caster_Wheel.stl"/>
            </geometry>
        </collision>
        
        <inertial>
            <mass value="0.001"/>
            <origin xyz="0 0 0" rpy="0 0 0"/>
            <inertia
            ixx="1e-6" ixy="0.0" ixz="0.0"
            iyy="1e-6" iyz="0.0"
            izz="1e-6"/>
        </inertial>
    </link>

    <joint name="base_link_joint" type="fixed">
        <parent link="base_footprint"/>
        <child link="base_link"/>
        <origin xyz="0. 0. 0.05" rpy="0. 0. 0."/>
    </joint>

    <joint name="wheel_l_joint" type="continuous">
        <parent link="base_link"/>
        <child link="wheel_l"/>
        <origin xyz="0.052 0.095 -0.0025" rpy="0. 0. 0."/>
        <axis xyz="0 1 0"/>
    </joint>

    <joint name="wheel_r_joint" type="continuous">
        <parent link="base_link"/>
        <child link="wheel_r"/>
        <origin xyz="0.052 -0.095 -0.0025" rpy="0. 0. 0."/>
        <axis xyz="0 1 0"/>
    </joint>

    <joint name="caster_joint" type="fixed">
        <parent link="base_link"/>
        <child link="caster"/>
        <origin xyz="-0.095 0. -0.03" rpy="0. 0. 0."/>
    </joint>
</robot>